pydantic/_internal/_generate_schema.py

```diff
@@ -279,12 +279,14 @@ def _generate_schema(self, obj: Any, from_dunder_get_core_schema: bool) -> core_
         """
         Recursively generate a pydantic-core schema for any supported python type.
         """
-        if isinstance(obj, str):
-            return {'type': obj}  # type: ignore[return-value,misc]
```

Comment:
For context, without the changes in this PR, this line was not covered in our current test suite (I replaced it with assert False and all tests were passing).